source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -135,8 +136,10 @@ void TcpStatsdSink::TlsSink::commonFlush(const std::string& name, uint64_t value
   // This written this way for maximum perf since with a large number of stats and at a high flush
   // rate this can become expensive.
   const char* snapped_current = current_slice_mem_;
-  memcpy(current_slice_mem_, STAT_PREFIX, sizeof(STAT_PREFIX) - 1);
-  current_slice_mem_ += sizeof(STAT_PREFIX) - 1;
+  memcpy(current_slice_mem_, parent_.getPrefix().c_str(), parent_.getPrefix().size());
+  current_slice_mem_ += parent_.getPrefix().size();
+  *current_slice_mem_ = '.';
```

Comment:
nit: simplify to `*current_slice_mem_++ = '.'` like below. Also, technically you should modify the above space reservation comment to account for this dot.